pkg/v3/hooks/prebuild.go

```diff
@@ -0,0 +1,29 @@
+package hooks
+
+import (
+	ocr2keepersv3 "github.com/smartcontractkit/ocr2keepers/pkg/v3"
+)
+
+type resultRemover interface {
+	Remove(...string)
+}
+
+func NewPrebuildHookRemoveFromStaging(remover resultRemover) *PrebuildHookRemoveFromStaging {
+	return &PrebuildHookRemoveFromStaging{remover: remover}
+}
+
+type PrebuildHookRemoveFromStaging struct {
+	remover resultRemover
+}
+
+func (hook *PrebuildHookRemoveFromStaging) RunHook(outcome ocr2keepersv3.AutomationOutcome) error {
+	toRemove := make([]string, 0, len(outcome.Performable))
+
+	for _, result := range outcome.Performable {
+		toRemove = append(toRemove, result.Payload.ID)
```

Comment:
I wonder what the performance difference would be in calling `hook.remover.Remove(result.Payload.ID)` directly instead of building a slice of IDs to be removed? 🤔

Reply:
That's a good question. That may be resolved in the result store. Maybe using multiple goroutines to remove values? Right now it just runs a for loop on the array of strings. So passing one string at a time just starts up a for loop each time with a single value. ¯\_(ツ)_/¯